src/librustc_mir/mir_map.rs

```diff
@@ -150,6 +143,32 @@ impl<'a, 'm, 'tcx> Visitor<'tcx> for InnerDump<'a,'m,'tcx> {
 
         intravisit::walk_fn(self, fk, decl, body, span);
     }
+
+    fn visit_item(&mut self, i: &'tcx hir::Item) {
+        match i.node {
+            hir::ItemConst(_, ref expr) |
+            hir::ItemStatic(_, _, ref expr) => {
+                let param_env = ty::ParameterEnvironment::for_item(self.tcx, i.id);
+                let infcx = infer::new_infer_ctxt(self.tcx,
+                                                  &self.tcx.tables,
+                                                  Some(param_env),
+                                                  ProjectionMode::AnyFinal);
+                let (mir, scope_auxiliary) = build::construct_expr(Cx::new(&infcx),
+                                                                   i.id,
+                                                                   i.span,
+                                                                   expr);
+                pretty::dump_mir(self.tcx,
+                                 "mir_map",
+                                 &0,
+                                 i.id,
+                                 &mir,
+                                 Some(&scope_auxiliary));
+                assert!(self.map.map.insert(i.id, mir).is_none());
```

Comment:
Yeah, a PR like this is a prerequisite for experimenting with replacing the current constant evaluator with Miri.

Currently Miri is stumped by statics and non-trivial constants since they are given by HIR that it can't interpret and there is no compiler-plugin-accessible way to lower HIR to MIR last time I checked.